test/python/WMCore_t/MicroService_t/MSPileup_t/MSPileup_t.py

```diff
@@ -0,0 +1,113 @@
+        """
+        cherrypy.request.user = "test"
+        self.validRSEs = ['rse1', 'rse2']
+        msConfig = {'reqmgr2Url': 'http://localhost',
+                    'rucioAccount': 'wmcore_mspileup',
+                    'rucioUrl': 'http://cms-rucio-int.cern.ch',
+                    'rucioAuthUrl': 'https://cms-rucio-auth-int.cern.ch',
+                    'mongoDB': 'msPileupDB',
+                    'mongoDBCollection': 'msPileupDBCollection',
+                    'mongoDBServer': 'mongodb://localhost',
+                    'mongoDBReplicaSet': '',
+                    'mongoDBUser': None,
+                    'mongoDBPassword': None,
+                    'validRSEs': self.validRSEs,
+                    'mockMongoDB': True}
+        self.mgr = MSPileup(msConfig)
+
+
+        self.pname = '/lksjdflksdjf/kljsdklfjsldfj/PREMIX'
+        expectedRSEs = self.validRSEs
+        fullReplicas = 0
+        campaigns = ['c1', 'c2']
+        data = {
+            'pileupName': self.pname,
+            'pileupType': 'classic',
+            'expectedRSEs': expectedRSEs,
+            'currentRSEs': expectedRSEs,
+            'fullReplicas': fullReplicas,
+            'campaigns': campaigns,
+            'containerFraction': 0.0,
+            'replicationGrouping': "ALL",
+            'active': True,
+            'pileupSize': 0,
+            'ruleList': []}
+
+        obj = MSPileupObj(data, validRSEs=self.validRSEs)
+        for key in ['insertTime', 'lastUpdateTime', 'activatedOn', 'deactivatedOn']:
+            self.assertNotEqual(obj.data[key], 0)
+        self.assertEqual(obj.data['expectedRSEs'], expectedRSEs)
+        self.assertEqual(obj.data['fullReplicas'], fullReplicas)
+        self.assertEqual(obj.data['campaigns'], campaigns)
+        self.doc = obj.getPileupData()
+
+    def testMSPileupHTTPApis(self):
+        "test MSPileup HTTP APIs"
+
+        # create doc
+        res = self.mgr.createPileup(self.doc)
+        self.assertEqual(len(res), 0)
+
+        # get doc
+        spec = {'pileupName': self.pname}
+        res = self.mgr.getPileup(**spec)
+        self.assertEqual(len(res), 1)
+        # self.assertDictEqual(res[0], self.doc)
+
+        # query doc
+        projection = ["pileupType"]
+        res = self.mgr.queryDatabase(spec, projection)
+        self.assertEqual(len(res), 1)
+        # self.assertEqual(res[0], ["classic"])
+
+        # update doc
+        doc = dict(self.doc)
+        doc["pileupType"] = "new"
+        res = self.mgr.updatePileup(doc)
+        self.assertEqual(len(res), 0)
+
+        # query doc
+        res = self.mgr.queryDatabase(spec, projection)
+        self.assertEqual(len(res), 1)
+        # self.assertEqual(res[0], ["new"])
+
+        # delete doc
+        res = self.mgr.deletePileup(spec)
+        self.assertEqual(len(res), 0)
+
+    def testMSPileupError(self):
+        "test mspileupError function"
+        doc = {'error': 'mspileup error', 'code': 123, 'message': 'msg'}
+        try:
```

Comment:
Please use a construction like:
```
        with self.assertRaises(cherrypy.HTTPError):
            mspileupError(doc)
```

otherwise we can't be sure whether this test is really testing that function.

In addition, please add a success case as well.

Reply:
done